### PR TITLE
Add support for Step 3.5 flash!

### PR DESCRIPTION
**Pull request by rltakashige**
## Motivation

Working version of #1366 

## Changes

Add Step 3.5 Flash

### Manual Testing
Works!

### Automated Testing
Running two processes tensor/pipeline sharded gives same logits as single process.

### CODE REVIEW

**Review by JakeHillion — APPROVED** [review]
LGTM.
